### PR TITLE
Add .cairo-coverage-ignore tests

### PR DESCRIPTION
**Pull request by ksew1**
**Stack**:
- #111
- #110 ⬅
- #109
- #108

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: crates/cairo-coverage/tests/data/coverage_ignore/Scarb.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+name = "coverage_ignore"
+version = "0.1.0"
+edition = "2024_07"
+
+# See more keys and their definitions at https://docs.swmansion.com/scarb/docs/reference/manifest.html
+[dependencies]
+starknet = ">=2.8.0"
+
+[dev-dependencies]
+snforge_std = { git = "https://github.com/foundry-rs/starknet-foundry", tag = "v0.29.0" }
```
Comment (THenry14) [inline crates/cairo-coverage/tests/data/coverage_ignore/Scarb.toml:11]: nit: can we use latest version? (you could use registry as well)

**Review by THenry14 — COMMENTED** [review]
it would be cool to also test the edge cases related to the https://github.com/software-mansion/cairo-coverage/pull/109/files (not necessarily e2e tests, could be just unit)

As an example: 
- correct errors are thrown for CairoCoverageIgnoreMatcher::new (file does not exist?)
- does find_ignore_file really work for things up to root?

etc

**Comment by ksew1** [comment]
> it would be cool to also test the edge cases related to the https://github.com/software-mansion/cairo-coverage/pull/109/files (not necessarily e2e tests, could be just unit)
> 
> As an example:
> 
>     * correct errors are thrown for CairoCoverageIgnoreMatcher::new (file does not exist?)
> 
>     * does find_ignore_file really work for things up to root?
> 
> 
> etc

Really good idea 😄. Added this

**Review by THenry14 — APPROVED** [review]
🫵🏻 🪨
